src/rpdk/core/contract/resource_generator.py

```diff
@@ -32,12 +33,12 @@
 # date is extraction from date-time
 # time is extraction from date-time
 STRING_FORMATS = {
-    "arn": "^arn:aws(-(cn|gov))?:[a-z-]+:(([a-z]+-)+[0-9])?:([0-9]{12})?:[^.]+$",
+    "arn": "^arn:aws(-(cn|us-gov))?:[a-z-]+:([a-z]{2}-([a-z]+-)+[0-9])+:([0-9]{12}){1}:[a-z0-9-:/*]+$",
```

Comment:
`([a-z]{2}-([a-z]+-)+[0-9])+` I think ? is more appropriate here? What am I missing?  Region can either be blank or a single region.

Reply:
`([0-9]{12}){1}` I think `?` is more accurate here.  There are valid ARNs that have no account IDs.
For example IAM AWS Managed policies have an ARN like `arn:aws:iam::aws:policy/AmazonSQSFullAccess`

Reply:
I was basing it off https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html i seem to have overlooked the global resources dont need the region.

Will fix.

Reply:
`[a-z0-9-:/*]+`  I think this is too restrictive too and I believe its missing a slash `[a-z0-9-:\/*]`

As an example a IAM role ARN typically includes the name:
`arn:aws:iam::account:role/role-name-with-path`
and from the documentation:
```
Names of users, groups, roles, policies, instance profiles, and server certificates must be alphanumeric, including the following common characters: plus (+), equal (=), comma (,), period (.), at (@), underscore (_), and hyphen (-).
```

I believe there is a unicode character thing here too.  I would make it more like what it was before.